plugins/index-by-search/src/main/java/org/elasticsearch/plugin/indexbysearch/IndexBySearchRequest.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.plugin.indexbysearch;
+
+import static java.lang.Math.min;
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+import static org.elasticsearch.search.sort.SortBuilders.fieldSort;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.index.IndexRequest.OpType;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.lucene.uid.Versions;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.index.VersionType;
+import org.elasticsearch.plugin.indexbysearch.AbstractAsyncScrollAction.AsyncScrollActionRequest;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+
+public class IndexBySearchRequest extends ActionRequest<IndexBySearchRequest> implements AsyncScrollActionRequest {
+    private static final TimeValue DEFAULT_SCROLL_TIMEOUT = TimeValue.timeValueMinutes(5);
+    private static final int DEFAULT_SIZE = 100;
```

Comment:
Its possible to do as part of the request. 100 is, admittedly, a pretty silly default. Its better than 10 though.